libs/api/ebpf_api.cpp

```diff
@@ -2865,6 +2874,72 @@ _Requires_lock_not_held_(_ebpf_state_mutex) _Must_inspect_result_ ebpf_result_t
 }
 CATCH_NO_MEMORY_EBPF_RESULT
 
+EBPF_API_LOCKING
+_Must_inspect_result_ ebpf_result_t
+ebpf_object_open_memory(
+    _In_reads_(buffer_size) const uint8_t* buffer,
+    size_t buffer_size,
+    _In_opt_z_ const char* object_name,
+    _In_opt_z_ const char* pin_root_path,
+    _In_opt_ const ebpf_program_type_t* program_type,
+    _In_opt_ const ebpf_attach_type_t* attach_type,
+    _Outptr_ struct bpf_object** object,
+    _Outptr_result_maybenull_z_ const char** error_message) NO_EXCEPT_TRY
+{
+    EBPF_LOG_ENTRY();
+    ebpf_assert(buffer);
+    ebpf_assert(object);
+    ebpf_assert(error_message);
+    *error_message = nullptr;
+
+    EBPF_LOG_MESSAGE(
+        EBPF_TRACELOG_LEVEL_INFO, EBPF_TRACELOG_KEYWORD_API, "ebpf_object_open_memory: loading from memory");
+
+    ebpf_object_t* new_object = new (std::nothrow) ebpf_object_t();
+    if (new_object == nullptr) {
+        EBPF_RETURN_RESULT(EBPF_NO_MEMORY);
+    }
+
+    set_global_program_and_attach_type(program_type, attach_type);
+
+    std::vector<uint8_t> data(buffer, buffer + buffer_size);
+
+    ebpf_result_t result =
+        _initialize_ebpf_object_from_file(data, object_name, pin_root_path, new_object, error_message);
+    if (result != EBPF_SUCCESS) {
+        std::string log_message = "ebpf_object_open_memory: error loading memory:";
+        log_message += ", error message:";
```

Comment:
nit: this is an error in the control path, no not critical, but could use string.append() instead since it's not test code

Reply:
Why prefer operator += over .append?